src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -509,7 +509,22 @@ private void createTableIfNotExists(final String tableName) {
       }
     } else {
       LOGGER.info("Creating new table {}.", tableName);
-      this.conn.createTable(tableName);
+      if (this.enableSchematization) {
+        if (Utils.usesAvroValueConverter(connectorConfig)) {
+          // if we could read schema from schema registry
+          this.conn.createTableWithSchema(
+              tableName,
+              Utils.getSchemaMapForTable(
+                  tableName,
+                  this.topicToTableMap,
+                  this.connectorConfig.get(
+                      SnowflakeSinkConnectorConfig.VALUE_SCHEMA_REGISTRY_CONFIG_FIELD)));
+        } else {
+          throw SnowflakeErrors.ERROR_5021.getException();
+        }
```

Comment:
It looks weird that we're throwing in else branch, should we throw it inside `usesAvroValueConverter`?